build-system/tasks/visual-diff/index.js

```diff
@@ -432,15 +433,26 @@ async function snapshotWebpages(percy, browser, webpages) {
       log('verbose', 'Navigating to page', colors.yellow(webpage.url));
 
       // Puppeteer is flaky when it comes to catching navigation requests, so
-      // ignore timeouts. If this was a real non-loading page, this will be
-      // caught in the resulting Percy build. Also attempt to wait until there
-      // are no more network requests. This method is flaky since Puppeteer
-      // doesn't always understand Chrome's network activity, so ignore timeouts
-      // again.
-      const pagePromise = page.goto(fullUrl, {waitUntil: 'networkidle0'})
+      // retry the page navigation up to 3 times (the `retries` parameter is the
+      // number of follow-up attempts, so the total is `retries`+1) and
+      // eventually ignore a final timeout. If this was ends up being a real
+      // non-loading page error, this will be caught in the resulting Percy
+      // build. Also attempt to wait until there are no more network requests.
+      // This method is flaky since Puppeteer doesn't always understand Chrome's
+      // network activity, so ignore timeouts again.
+      const pagePromise = promiseRetry({retries: 2}, async(retry, number) => {
```

Comment:
Nit: if you're using async/await, a retry library really isn't necessary.

```js
async function goto(url, retries = 1) {
  let lastError;
  for (let i = 0; i < retries; i++) {
    try {
      return await page.goto(url);
    } catch (e) {
      lastError = e;
    }
  }
  throw lastError;
}

Reply:
Oh right, makes sense. Less dependencies 😄